src/main/asciidoc/index.adoc

```diff
@@ -17,503 +17,801 @@
+====
 
-* Prepare your HTTP message using the JWS.
-** The access tokens in OAuth 2.0 are of type *bearer*, meaning the client needs to pass the token with each request.
-** The followings are the expected HTTP headers:
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+The authorization server MUST be configured so that it returns access tokens that are self-contained in lieu of opaque
+tokens.
+====
 
-[width="100%",cols="22%,19%,17%,42%",]
-|===
-|*HTTP Header* |*Description* |*Required* |*Value*
-|Authorization |Must be set to the access token. |Must |Bearer \{Access Token}
-|Content-Type |Content type of the payload. |Optional a|
-\{JSON body} -> application/jose+json
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+Indeed, opaque tokens MUST be validated at the authorization server's token introspection endpoint for each request,
+which is grossly inefficient.
+====
 
-\{Non JSON body} -> application/jose
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+Self-contained tokens, by contrast, SHOULD be validated locally by the resource server, assuming that there is a
+pre-established trust relationship between the resource server and the authorization server.
+====
 
-|===
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+For RESTful services processing Category 3-4 data, it's RECOMMENDED to validate all access tokens at the authorization
+server's token introspection endpoint (despite the performance penalty).
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+As any other confidential, access tokens MUST be kept confidential in transit and at rest, and only shared among the
+authorization server, the resource servers the access token is valid for (in a micro-services architecture, access
+tokens MAY be propagated between services), and the client to whom the access token is issued.
+====
+
+Self-contained tokens contain a set of claims stored in the header part of an
+https://datatracker.ietf.org/doc/html/rfc7519[RFC 7519] compliant JSON Web Token (JWT; pronounced "jot") object
+represented as either a https://datatracker.ietf.org/doc/html/rfc7515[JSON Web Signature (JWS)] and/or
+https://datatracker.ietf.org/doc/html/rfc7516[JSON Web Encryption (JWE)] structure.
+
+// TODO: decide whether we should provide an in-depth description or simply a link to the RFC
+
+A claim is simply a key/value pair holding some piece of information asserted about an entity that can be used for
+access control decisions.
+
+The JWT standard distinguishes between reserved claims and custom claims (public or private).
+
+Reserved claims are standardized claims that have been defined to enable interoperability.
+A full list of reserved claims can also be found on the https://www.iana.org/assignments/jwt/jwt.xhtml[IANA JSON Web
+Token Claims Registry].
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+A resource server SHOULD validate the reserved claims provided in an access token’s header.
```

Comment:
A resource server SHOULD validate the reserved claims provided in an access token’s **payload**.